test/parallel_exec.jl

```diff
@@ -774,6 +774,17 @@ end
 # Test pmap with a generator type iterator
 @test [1:100...] == pmap(x->x, Base.Generator(x->(sleep(0.0001); x), 1:100))
 
+# Test pgenerate
+a = rand(4,4)
+b = fill(a, 4)
+c = collect(Base.pgenerate(svdfact, b))
```

Comment:
As `pgenerate` is expected to process elements a) asynchronously and b) return them in the correct order, I would improve this test by incorporating a random delay:

`c = collect(Base.pgenerate(x->(sleep(rand()*0.1);svdfact(x);, b))`